### PR TITLE
Improve KTooltip documentation

### LINKED ISSUE #694: Add documentation to the KTooltip page
🌱 **Are you new to the codebase? Welcome! Please see [the contributing guidelines](https://github.com/learningequality/kolibri-design-system/blob/develop/CONTRIBUTING.md).**

## Summary

The [`KTooltip` documentation page](https://design-system.learningequality.org/ktooltip/) lacks an overview and examples of usage. The goal of this issue is to:

- [ ]  Add the “Overview” section with a brief introduction on what the component is for.
- [ ]  Add the “Usage” section with sub-sections documenting:
    - [ ]  The basic usage with an example code snippet and the visualization of its output. The example should include `reference` and `refs` explanation and show how `KTooltip` is used together with another component, such as `KIcon` ([Kolibri codebase examples](https://github.com/search?q=repo%3Alearningequality%2Fkolibri%20%3CKTooltip&type=code)).
    - [ ]  A visualization of the `disabled` state
    - [ ]  A visualization of all supported `placement`s
    - [ ]  A visualization of `maxWidth`

## Guidance

- For consistency in documentation, first see the [“Writing” guidelines](https://design-system.learningequality.org/writing/) and use existing pages for reference, such as the [`KTextbox`](https://design-system.learningequality.org/ktooltip/) [documentation](https://design-system.learningequality.org/ktextbox/) and [`KImg` documentation](https://design-system.learningequality.org/kimg/) pages.
- Generally we try to strike a balance between documenting all important modes of a component, while not overwhelming readers with unnecessary details or exhaustive lists of all possible states.
- Use simple, clear, and concise language. Think about whether what you are explaining can be easily understood by newcomers.
- To understand the technical aspects of writing documentation, refer to:
    - [“How to update the documentation website”](https://github.com/learningequality/kolibri-design-system/blob/release-v4/dev_docs/04_how_to_update_docs.md)
    - Documentation pages code, such as [`KTextbox` docs source code](https://github.com/learningequality/kolibri-design-system/blob/9d2a147de7e81c945ca0eab7ffb2c4719fd2bb06/docs/pages/ktextbox.vue) or [`KImg`'s docs source code](https://github.com/learningequality/kolibri-design-system/blob/9d2a147de7e81c945ca0eab7ffb2c4719fd2bb06/docs/pages/kimg.vue)

### PR DESCRIPTION
**Pull request by shruti862**
## Description
I added Overview and Usage section on [KTooltip documentation page](https://design-system.learningequality.org/ktooltip/)

#### Issue addressed
Closes #694

Addresses PR #853

## Changelog

  - **Description:** Add documentation to the `KTooltip` page
  - **Products impact:** none
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/694
  - **Components:** `KTooltip`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -

## Steps to test

1. Step 1
2. Step 2
3. ...

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [ ] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

### CODE REVIEW

File: docs/pages/ktooltip.vue [hunk 1]
```diff
@@ -1,6 +1,101 @@
 <template>
 
-  <DocsPageTemplate apiDocs />
+  <DocsPageTemplate apiDocs>
+    <DocsPageSection title="Overview" anchor="#overview">
+      The KTooltip component displays a tooltip when a user hovers over a specific element.
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
      Displays a tooltip when a user hovers over a specific element.
```

File: docs/pages/ktooltip.vue [hunk 2]
```diff
@@ -1,6 +1,101 @@
...
+      <DocsShowCode language="html">
+        <KTooltip reference="myButton" :refs="$refs">
+          Click me to learn more!
+        </KTooltip>
+        <button ref="myButton">
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        In this example, the <code>ref</code> attribute is used to create a reference to the
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
        The <code>ref</code> attribute creates a reference to the 
```

File: docs/pages/ktooltip.vue [hunk 3]
```diff
@@ -1,6 +1,101 @@
...
+        <KTooltip reference="myButton" :refs="$refs">
+          Click me to learn more!
+        </KTooltip>
+        <button ref="myButton">
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        In this example, the <code>ref</code> attribute is used to create a reference to the 
+        <code>myButton</code> element, and the <code>reference</code> prop is used to associate the
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
        <code>myButton</code> element, and the <code>reference</code> prop associates the
```

File: docs/pages/ktooltip.vue [hunk 4]
```diff
@@ -1,6 +1,101 @@
...
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        In this example, the <code>ref</code> attribute is used to create a reference to the 
+        <code>myButton</code> element, and the <code>reference</code> prop is used to associate the 
+        tooltip with this reference.
+      </p>
+
+      <h3>Disabled State</h3>
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
      <h3>Disabled state</h3>
```

File: docs/pages/ktooltip.vue [hunk 5]
```diff
@@ -1,6 +1,101 @@
...
+      <DocsShowCode language="html">
+        <KTooltip reference="disableButton" :refs="$refs" disabled>
+          This tooltip is disabled
+        </KTooltip>
+        <button ref="disableButton">
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        This example demonstrates the disabled state of the KTooltip component. The
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```
<p>The <code>disabled</code> prop can be used to disable the tooltip.</p>
```

File: docs/pages/ktooltip.vue [hunk 6]
```diff
@@ -1,6 +1,101 @@
...
+        <button ref="disableButton">
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        This example demonstrates the disabled state of the KTooltip component. The 
+        <code>disabled</code> prop is set to <code>true</code> to disable the tooltip.
+      </p>
+
+      <h3>Supported Placements</h3>
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
      <h3>Supported placements</h3>
```

File: docs/pages/ktooltip.vue [hunk 7]
```diff
@@ -1,6 +1,101 @@
...
+        <button ref="topButton">
+          Top
+        </button>
+      </DocsShowCode>
+      <p>
+        This example demonstrates the supported placements of the KTooltip component. The 
+        <code>placement</code> prop is used to specify the placement of the tooltip.
+      </p>
+
+      <h3>Max Width</h3>
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```suggestion
      <h3>Maximum width</h3>
```

File: docs/pages/ktooltip.vue [hunk 8]
```diff
@@ -1,6 +1,101 @@
...
+      <DocsShowCode language="html">
+        <KTooltip reference="maxWidthButton" :refs="$refs" maxWidth="200px">
+          This tooltip has a max width of 200px
+        </KTooltip>
+        <button ref="maxWidthButton">
+          Learn More
+        </button>
+      </DocsShowCode>
+      <p>
+        This example demonstrates the <code>maxWidth</code> prop of the KTooltip component.
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```
<p>The <code>maxWidth</code> prop can be used to limit the tooltip width.</p>
```

File: docs/pages/ktooltip.vue [hunk 9]
```diff
@@ -1,6 +1,101 @@
...
+      </div>
+      <DocsShowCode language="html">
+        <KTooltip reference="topButton" :refs="$refs" placement="top">
+          Top placement
+        </KTooltip>
+        <button ref="topButton">
+          Top
+        </button>
+      </DocsShowCode>
+      <p>
```
Comment (MisRob) [inline docs/pages/ktooltip.vue]:
```
<p>The <code>placement</code> prop can be used to specify the placement of the tooltip.</p>
```

**Review by MisRob — CHANGES_REQUESTED** [review]
Thank you for getting this ready @shruti862. Overall looks great and describes all scenarios requested in the issue. I suggest few tweaks regarding casing, conciseness, and clarity ([writing guide](https://design-system.learningequality.org/writing/) has more information). You can submit most of them here in GitHub UI if you'd like.

**Comment by shruti862** [comment]
@MisRob I have done all the suggested changes.

**Review by MisRob — APPROVED** [review]
Thanks @shruti862! I pulled in the latest `develop` and re-linted because of our recent Node version upgrade.

**Pull request merged**